src/OrbitGl/GlCanvas.cpp

```diff
@@ -30,15 +33,15 @@ float GlCanvas::kZValueBox = 0.05f;
 float GlCanvas::kZValueEvent = 0.07f;
 float GlCanvas::kZValueTrackText = 0.08f;
 float GlCanvas::kZValueTrackLabel = 0.09f;
+float GlCanvas::kZValueTimeBar = 0.31f;
```

Comment:
nit. Instead of arbitrarily assigning z-values, I wonder if we could use an enum. We could compute the z-value from the enum value. Something like:

```
enum class DrawableType { Event, TrackText, TrackLabel, TimeBar, ... };

[[nodiscard]] float GetZValue(DrawableType type) {
   return static_cast<float>(type) * 0.1f;
}

```

We could cache the z-values in a look up table too for faster access.

Adding a new type would only require modifying the enum, not the GlCanvas.cpp file, and we wouldn't have to assign an arbitrary floating point value. The order of the enum elements would determine the relative z-ordering of the different types. Wdyt?

Reply:
I like the solution. It makes me remember the discussion we started 1 year and half in https://github.com/google/orbit/pull/1406. I also don't like to use floats here.

There are 2 issues that after discussing with @reichlfl, leads us to not make the enum at that time:
1- We are using a few non-listed values (using +- epsilons to the z-value) to draw sliders that needs to be refactored . This should be easy to solve and actually we should make it, because it's completely hidden inside sliders.
2- We have 2 features, Moving Tracks and Pinning Tracks, that are moving a set of z-values to the front. In our case, the real values should be: { Event, TrackText, TrackLabel, MovingTrackEvent, MovingTrackText, MovingTrackLabel, PinnedTrackEvent, PinnedTrackText, PinnedTrackLabel, ... }. In case of adding a new layer on the track, we will need to add 3 different new layers. This is more difficult to solve.
Possible solutions:
a - Design MovingTrack and PinnedTrack differently (how?), so we can use enum in the z-layers.
b - Change our way to draw, so the parent is the one that decides which children should be on top of each other. 
   i) We require to have all the TimeGraph in a tree (where TimeGraph is the root). This is in process. I'm working with the sliders, and only the green mouse line and the overlay are missing.
   ii) We also require that every child and its descendant should have a strict order with all other children. I mean, Sliders > Overlay > Timeline > TrackContainer (children of TimeGraph) and IteratorsBoxes > PinnedTrack > MovingTrack > Track (Children of TrackContainer). This is as far as I know something that we are having, but we cannot set the Overlay in between of TimelineTicks and TimelineBackground as Carsten wanted for example.
c- Leave everything as it is.

Our current plan with @reichlfl is (c) until (b) is doable. I don't have a real plan to do (b) but I think we only should be aware of following the order which events are inserted also for different kind of shapes (we cannot draw first all boxes and after all lines, etc).

What do you think? @pierricgimmig @vickyliu-go4it 
@reichlfl for visibility

Reply:
I like the idea of enum since we don't need to update the z values in the cpp file anymore. Also, it makes the order of different layers much clear.
But I think this could be done in another change. 

> 1- We are using a few non-listed values (using +- epsilons to the z-value) to draw sliders that needs to be refactored . This should be easy to solve and actually we should make it, because it's completely hidden inside sliders.

For the point 1, I would like to know what are the cases that we use "+- epsilons to the z-value". Could these cases map to one or several properly named z values?

> 2b- ... Design MovingTrack and PinnedTrack differently (how?)

How do we handle this two cases for now? It would be great if we can draw the track differently according to its state (i.e., moving or pinned). But if now we are handling them with the same z-values, then it would still be fine if we assign the same enum value for them for now.
We can still add more cases to the enum when (b) is doable. It does not hurt.

But if we are planning for (c) already, then I agree that it's unnecessary to put too much effort here for now.

Reply:
> For the point 1, I would like to know what are the cases that we use "+- epsilons to the z-value". Could these cases map to one or several properly named z values?

TBH, I've never read carefully that code. I'm planning to refactor it soon and I will have a better idea. But yes, kin any case we just simply will need a few extra z-values.

> How do we handle this two cases for now?  It would be great if we can draw the track differently according to its state (i.e., moving or pinned). But if now we are handling them with the same z-values, then it would still be fine if we assign the same enum value for them for now.

It's below in the code, we are just adding +0.1 or +0.2 to each z-layer, so we are NOT handling them with the same z-value. We are indirectly using different ones.

> We can still add more cases to the enum when (b) is doable. It does not hurt.

You are meaning option (a) here?

Reply:
Ah, I just saw these two comment lines for "tracks - moving" and "tracks - pinned".

If that is the case, I prefer to add enum cases for them both. Adding 0.1~0.2 sounds quite hacky.

Reply:
Can't everything stay the same except for the way we define and access the z-values? For pinned and moving tracks for example, we could still apply an offset relative to the value we get from a "GetZValue(DrawableType type)" function, no? What I was suggesting was a way to programmatically compute the z-value based on an enum, but the value is still a floating point that we can use as before.

Reply:
Something like replacing 
`float text_z = GlCanvas::kZValueTrackText;` by 
`float text_z = GetZValue(GlCanvas::kTrackText);`

Reply:
Feel free to merge the PR as is btw, this is a bit out of scope.

Reply:
I've made this bug: http://b/227341686